core/src/Kokkos_ExecPolicy.hpp

```diff
@@ -249,45 +249,48 @@ class RangePolicy : public Impl::PolicyTraits<Properties...> {
   // To be replaced with std::in_range (c++20)
   template <typename IndexType>
   static void check_conversion_safety(const IndexType bound) {
+    if constexpr (std::is_convertible_v<member_type, IndexType>) {
```

Comment:
Please add a comment above this line, something like
"Checking that the round-trip conversion preserve the value"

Reply:
Added a comment.